src/components/Layout/Layout.component.jsx

```diff
@@ -1,9 +1,40 @@
-import React from 'react';
-
+import React, { useLayoutEffect, useContext, useRef } from 'react';
+import { random } from '../../utils/fns';
+import { AppContext } from '../../utils/AppContext.provider';
 import './Layout.styles.css';
 
 function Layout({ children }) {
-  return <main className="container">{children}</main>;
+  const { state } = useContext(AppContext);
+  const { darkMode } = state;
+  const mainPage = useRef();
+
+  useLayoutEffect(() => {
+    const page = mainPage.current;
+
+    function rotateBackground() {
+      const xPercent = random(100);
+      const yPercent = random(100);
+      page.style.setProperty('--bg-position', `${xPercent}% ${yPercent}%`);
+    }
+    const intervalId = setInterval(rotateBackground, 3000);
+    page.addEventListener('click', rotateBackground);
+
+    return () => {
+      clearInterval(intervalId);
+      page.removeEventListener('click', rotateBackground);
+    };
+  }, []);
```

Comment:
```js
  useLayoutEffect(() => {
    const page = mainPage.current;

    function rotateBackground() {
      const xPercent = random(100);
      const yPercent = random(100);
      page.style.setProperty('--bg-position', `${xPercent}% ${yPercent}%`);
    }
    const intervalId = setInterval(rotateBackground, 3000);
    page.addEventListener('click', rotateBackground);

    return () => {
      clearInterval(intervalId);
      page.removeEventListener('click', rotateBackground);
    };
  }, []);
```

Good logic, it's ok. Try to abstract this as a way of separate this logic from the component